lib/IDE/Refactoring.cpp

```diff
@@ -1453,6 +1456,105 @@ bool RefactoringActionExtractRepeatedExpr::performChange() {
                                           EditConsumer).performChange();
 }
 
+/// Compute a context C such that C is a parent context of A and B.
+/// If no such context exists, return \c nullptr.
+DeclContext *getCommonParentContext(DeclContext *A, DeclContext *B) {
```

Comment:
Would this be interesting enough to promote it to DeclContext.h?

Reply:
That's a good idea. But similarly, please put the implementation in a .cpp file.